src/app/pages/ArticlePage/ArticlePage.jsx

```diff
@@ -60,6 +60,7 @@ import {
   getMentions,
   getLang,
 } from '#lib/utilities/parseAssetData';
+import getAuthorTwitterHandle from '#lib/utilities/getAuthorTwitterHandle';
```

Comment:
I considered declaring `getAuthorTwitterHandle` in the following modules 
when defining the logic to extract the twitter handle.
1. src/app/pages/ArticlePage/ArticlePage.jsx
2. src/app/lib/utilities/parseAssetData/index.js

While (1) would be adherent to colocation, it posed the drawback of
increasing the
- size of ArticlePage with regards to number of lines
- responsibilities of the component
thereby affecting readability and the single responsibility principle.

With (2), the functions that exist in the module serve to extract
data from the CMS payload. There are no rules applied to the
extracted data. For this reason, `getAuthorTwitterHandle`, would not fit
the responsibility of `src/app/lib/utilities/parseAssetData/index.js` since
there are rules that need to be applied to the byline block.

Upon evaluating the colocation principle against the drawbacks above,
I decided to declare `getAuthorTwitterHandle` in its own module
- in keeping with the single responsibility principle,
- to conform to the use of utilities such as [these](https://github.com/bbc/simorgh/blob/fe9c593d84872cf70b198f6e74ad240290dcef09/src/app/pages/ArticlePage/ArticlePage.jsx#L52-L62) to provide values required by the ArticlePage component, and
- to ensure the ArticlePage remains small and readable by leveraging auxiliary
helpers/utilities.

Reply:
Do let me know if the grounds above, for the decision to home `getAuthorTwitterHandle`
in its own utility are permissible in relation to the colocation rule.
Happy to move the function into `src/app/pages/ArticlePage/ArticlePage.jsx`
if the separation function into a dedicated module is deemed untenable.
Thank you. 😄

Reply:
hey @alex-magana , thanks your really detailed reply!

My view here was the seperate module was perfectly reasonable, just its location in a general `utilities` directory was less than ideal as the module is only needed by the `ArticlePage`. So a seperate module under `src/app/pages/ArticlePage/` would be more desirable in my view.

Now its a really good point that the [existing Article utilities](https://github.com/bbc/simorgh/blob/fe9c593d84872cf70b198f6e74ad240290dcef09/src/app/pages/ArticlePage/ArticlePage.jsx#L52-L62) are also found in the `src/app/lib/utlities/` directory, I'd say though that that is going against our newly agreed coding standards. With this in mind, I'd maybe look at moving them under `src/app/pages/ArticlePage/` in a follow-up PR if they are only used in the `ArticlePage` and not elsewhere.

Where possible we want to apply our new standards as an [opportunistic refactor](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/React-Components.md#applying-standardisation) but it's very sensible to do this over multiple PRs so the functional change you are making is easy to review and PRs don't get too big.

For this PR maybe best to keep things consistent and keep it under `utilities` for now?

Reply:
Thanks for providing such extensive context Andrew.
Happy to raise a ticket fix forward, colocating of [parseAssetData](https://github.com/bbc/simorgh/blob/latest/src/app/lib/utilities/parseAssetData/index.js
).
I'll raise a ticket to address this.
As regards, `getAuthorTwitterHandle`, I've taken the [opportunistic refactor](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/React-Components.md#applying-standardisation)
approach and moved the helper to [src/app/pages/ArticlePage](https://github.com/bbc/simorgh/tree/75fa4327e118ca589b606cb1878c62e0fe216a2e/src/app/pages/ArticlePage/getAuthorTwitterHandle).